From 383a70a87049ef5065bba4c2c4d4bc3cff914358 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 23 Jun 2020 08:39:44 +0800
Subject: [PATCH] policy/modules/system/getty: allow getty_t watch
 getty_runtime_t file

Fixes:
type=AVC msg=audit(1592813140.280:26): avc:  denied  { watch } for
pid=385 comm="getty" path="/run/agetty.reload" dev="tmpfs" ino=12247
scontext=system_u:system_r:getty_t
tcontext=system_u:object_r:getty_runtime_t tclass=file permissive=1

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/getty.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/getty.te b/policy/modules/system/getty.te
index f5316c30a..39e27e5f1 100644
--- a/policy/modules/system/getty.te
+++ b/policy/modules/system/getty.te
@@ -47,6 +47,7 @@ allow getty_t getty_log_t:file { append_file_perms create_file_perms setattr_fil
 logging_log_filetrans(getty_t, getty_log_t, file)
 
 allow getty_t getty_runtime_t:dir watch;
+allow getty_t getty_runtime_t:file watch;
 manage_files_pattern(getty_t, getty_runtime_t, getty_runtime_t)
 files_runtime_filetrans(getty_t, getty_runtime_t, file)
 
@@ -65,6 +66,7 @@ dev_read_sysfs(getty_t)
 files_read_etc_runtime_files(getty_t)
 files_read_etc_files(getty_t)
 files_search_spool(getty_t)
+fs_search_tmpfs(getty_t)
 
 fs_search_auto_mountpoints(getty_t)
 # for error condition handling
-- 
2.17.1

